cpp/src/arrow/json/converter_test.cc

```diff
@@ -190,9 +190,12 @@ TEST(ConverterTest, Decimal128And256) {
     options.explicit_schema = schema({field("", decimal_type)});
 
     std::string json_source = R"(
-    {"" : "02.0000000000"}
-    {"" : "30.0000000000"}
-  )";
+      {"" : "02.0000000000"}
+      {"" : "30.0000000000"}
+      {"" : "30.01"}
+      {"" : "30.0000000000123"}
```

Comment:
Do we want truncation to happen silently or would we rather get an error here?

Reply:
For example, our CSV reader would emit an error instead of dropping some digits like this.

Reply:
Yeah, I think users would like different behaviors. E.g. in Impala, we'd like Arrow to continue reading the remaining rows instead of throwing an error and stop.

I think we can add an option to choose the behavior.

Reply:
I realized that the error handling in other converters are also stop and return errors, e.g.
https://github.com/apache/arrow/blob/d67a210b8c50b1d109e3d7780591d010e94cc9cc/cpp/src/arrow/json/converter.cc#L132-L134

So I tend to the same behavior now. I think we can add the option for configurable error handling behavior in a new JIRA.

Reply:
Thanks. Feel free to ping once that is done. Hopefully we can get fixed in time for 10.0.0.